File: c/matriz_geral.c
#include <stdio.h>
#include <math.h>

int main(){
    int n, indlinha, indcoluna;
    double somapositivos;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &n);

    double matriz[n][n];

    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%lf", &matriz[i][j]);
        }
    }

    somapositivos = 0;
    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
            if (matriz[i][j] > 0) {
                somapositivos = somapositivos + matriz[i][j];
            }
        }
    }

    printf("\nSOMA DOS POSITIVOS: %.1lf\n\n", somapositivos);

	printf("Escolha uma linha: ");
	scanf("%d", &indlinha);

	printf("LINHA ESCOLHIDA: ");

    for (int i=0; i<n; i++) {
        printf("%.1lf ", matriz[indlinha][i]);
    }

	printf("\n\nEscolha uma coluna: ");
	scanf("%d", &indcoluna);

    printf("COLUNA ESCOLHIDA: ");

    for (int i=0; i<n; i++) {
        printf("%.1lf ", matriz[i][indcoluna]);
    }

	printf("\n\nDIAGONAL PRINCIPAL: ");

    for (int i=0; i<n; i++) {
        printf("%.1lf ", matriz[i][i]);
    }

    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
            if (matriz[i][j] < 0) {
                matriz[i][j] = pow(matriz[i][j], 2);
            }
        }
    }

	printf("\n\nMATRIZ ALTERADA:\n");

    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
            printf("%.1lf ", matriz[i][j]);
        }
		printf("\n");
    }

    return 0;
}

File: python/matriz_geral.py
import math

n: int; indlinha: int; indcoluna: int
somapositivos: float

n = int(input("Qual a ordem da matriz? "))

matriz: [[float]] = [[0 for x in range(n)] for x in range(n)]

for i in range(n):
	for j in range(n):
		matriz[i][j] = float(input(f"Elemento [{i},{j}]: "))

somapositivos = 0
for i in range(n):
	for j in range(n):
		if matriz[i][j] > 0:
			somapositivos = somapositivos + matriz[i][j]

print(f"\nSOMA DOS POSITIVOS: {somapositivos:.1f}\n")

indlinha = int(input("Escolha uma linha: "))

print("LINHA ESCOLHIDA: ", end="");

for i in range(n):
	print(f"{matriz[indlinha][i]:.1f} ", end="")

indlinha = int(input("\n\nEscolha uma coluna: "))

print("COLUNA ESCOLHIDA: ", end="");

for i in range(n):
	print(f"{matriz[i][indlinha]:.1f} ", end="")

print("\n\nDIAGONAL PRINCIPAL: ", end="");

for i in range(n):
	print(f"{matriz[i][i]:.1f} ", end="")

for i in range(n):
	for j in range(n):
		if matriz[i][j] < 0:
			matriz[i][j] = math.pow(matriz[i][j], 2);

print("\n\nMATRIZ ALTERADA:");

for i in range(n):
	for j in range(n):
		print(f"{matriz[i][j]:.1f} ", end="")
	print()